### ISSUE TITLE
stagger /  shift time 

### ISSUE THREAD

**Issue by dcjona** [PlanetaryMotion/pingPongBallClock#52]
Hi, i see something stranger to resume : 
server is on 24/24, when i go to sleep i send this request : 
```
curl "http://ip:5000/api/setcontent" \
  -X POST \
  -d "content=time&checked=false" \
  -H "Content-Type: application/x-www-form-urlencoded" 
```

the next day this one : 
```
curl "http://ip:5000/api/setcontent" \
  -X POST \
  -d "content=time&checked=true" \
  -H "Content-Type: application/x-www-form-urlencoded"
``` 
the problem is that the time is shift (stagger) on the right. so i don't see the minute..

maybee a known bug?

i need to reboot the raspberry pi to correct the bug

**Comment by lancewilhelm** [comment]
Can you try and use the latest master branch and see if the bug persists. 

But, just by looking at what you have sent to the clock, you are essentially turning off the time before you go to bed, then you are turning on the time the next day, is that correct? What other settings do you have? Is it scrolling, or static? 

Soon the master branch will have a reset button to reset the screen and this may help solve this issue.

**Comment by lancewilhelm** [comment]
I have not seen this bug yet where it only displays the hour, is that correct? You are saying that the hours are where the minutes are usually? 

I will work on creating debug logging to help track down issues like this.

**Comment by dcjona** [comment]
yes correct i would like to turn off the clock when i go to bed. 
will check with the last master branch

**Comment by lancewilhelm** [comment]
> yes correct i would like to turn off the clock when i go to bed.

OK, please try and replicate the issue and give me the full breakdown of the settings. You can possibly provide me with a saved settings file. Also, if you use the latest master branch I will try to replicate on there. 

As a work around, you can just set the text color to be Red = 0, Green = 0, Blue = 0 when you go the bed which effectively turns off the clock. What it really does is turn off the LEDs but the code still thinks it is displaying text on the screen. This should solve your problem initially. Check the flaskUtils code and the javascript file to see how I send the RGB custom text color values. 

We still need to address your issue though so that it doesn't come up again. It may be indicating a larger issue.

**Comment by dcjona** [comment]
ok thanks for the answer i will try this instaed of timeoff

**Comment by neek0la** [comment]
I too would like to turn the clock off at bed time. 

Whats the possibility of a cronjob to set the colors to black in the evening and then reset to rainbow in the morning?

**Comment by lancewilhelm** [comment]
I am not sure that a straight cronjob running some function would be the easiest way to turn the colors to black in the evening. I think it is possible to do via a cronjob, but because the clock reacts to web requests it might be a little tricky. You may have to find something that sends a request at a certain time. 

I haven't been able to play with this much lately. I can potentially take a look at this and other bugs by the end of the week. I encourage you to try and send custom web requests to the clock. Take a look at the google chrome inspector Network tab when you click a button to send a color. This may give you some sense as to how it works. Then you can replicate that using a different piece of software that could be coded to work on a time basis.
